src/v2/shim_layer/controller_runtime.rs

```diff
@@ -385,6 +408,157 @@ where
     return Ok(Action::requeue(Duration::from_secs(60)));
 }
 
+// transactional_get_then_delete_by_retry retries get and then delete upon conflict errors to simulate atomic operations.
```

Comment:
More of a basic Kubernetes question than a comment on this code, but do real-life examples of this 'read->update' loop have a timeout, or would they just contend for a resource forever?

Reply:
They limit the number of retries (and also have back-off when a try fails).

Reply:
See this https://github.com/kubernetes/kubernetes/blob/0e64c6443f8e1f760c92a64304925986d4519a77/staging/src/k8s.io/client-go/util/retry/util.go#L68

Reply:
Is there some other way that would be facilitated? Or are we content, as a simplifier for the purposes of Anvil, with possible starvation?

Reply:
If we limit the number of retries then there is no way to have a refinement mapping from the retry code to the transactional spec API.

We could implement back-off later, which does not affect the liveness argument.

> with possible starvation?

What do you mean by starvation here? You mean the controller always retries and never terminates?

Reply:
Understandable, I was just curious.

I meant 'always retries'.